stan/math/gpu/kernels/lower_tri_inverse_step1.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef STAN_MATH_GPU_KERNELS_LOWER_TRI_INVERSE_STEP1_HPP
```

Comment:
I will make a more detailed document that will more precisely explain what is going on in each kernel. You are right regarding the names. I had trouble giving them simple descriptive names. 

The first step is actually a batched lower triangular inverse (we do N inverses of submatrices of size). Using the term batched lower triangular inverse might not be good, because we expect the submatrices to be on the diagonal of the input matrix, so its not a "general" batched inverse.

The second and third step perform special cases of batched matrix multiplication. If you look at slide 6 [here](https://github.com/rok-cesnovar/StanGPU-Workshop/blob/master/slides/Stan_%20GPU.pdf). 
Step 2 and 3 perform the calculation of C3. In the simple case like on slide 6 its 2 matrix multiplications and multiplying with -1. However when we have a larger matrix, like in slide 7. For the matrices marked with (2) we have to perform 4 smaller steps 2 and 3. Thats why I termed it batched.

Hopefully that gives some idea what is done here. Should I just make a Gist or just post it here?